lightning/src/chain/channelmonitor.rs

```diff
@@ -1923,7 +1940,22 @@ impl<Signer: Sign> ChannelMonitorImpl<Signer> {
 		}
 	}
 
-	pub(crate) fn broadcast_latest_holder_commitment_txn<B: Deref, L: Deref>(&mut self, broadcaster: &B, logger: &L)
+	pub(crate) fn maybe_broadcast_latest_holder_commitment_txn<B: Deref, L: Deref>(&mut self, broadcaster: &B, logger: &L)
+		where B::Target: BroadcasterInterface,
+					L::Target: Logger,
+	{
+		if self.allow_automated_broadcast{
+			for tx in self.get_latest_holder_commitment_txn(logger).iter() {
+				log_info!(logger, "Broadcasting local {}", log_tx!(tx));
+				broadcaster.broadcast_transaction(tx);
+			}
+			self.pending_monitor_events.push(MonitorEvent::CommitmentTxConfirmed(self.funding_info.0));
+		} else {
+			log_error!(logger, "You have a toxic holder commitment transaction avaible in channel monitor, read comment in ChannelMonitor::get_latest_holder_commitment_txn to be informed of manual action to take");
+		}
+	}
+
+	pub(crate) fn force_broadcast_latest_holder_commitment_txn_unsafe<B: Deref, L: Deref>(&mut self, broadcaster: &B, logger: &L)
```

Comment:
I don't think so, I mostly followed the signature for the previous function which was `pub(crate)`. But since both are supposed to be used through their respective wrappers, I guess we can just remove it? I did so in [c262be2](https://github.com/lightningdevkit/rust-lightning/pull/1593/commits/c262be27e6e73f819ebb3d94d4ddf084793e01c4)